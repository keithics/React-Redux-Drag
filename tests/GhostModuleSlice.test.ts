import {
  moduleInitialState,
  ModulePositionReducer,
  setPosition,
  startDrag,
} from '../src/redux/slices/ModulePositionSlice';

describe('Module Tests', () => {
  test('should return the initial state', () => {
    // @ts-ignore
    expect(ModulePositionReducer(undefined, {})).toMatchSnapshot();
  });

  test('should be able to change the module coordinates', () => {
    expect(
      ModulePositionReducer(
        moduleInitialState,
        setPosition({
          id: 2,
          coord: {
            x: 1,
            y: 2,
            h: 100,
            w: 2,
          },
        })
      )
    ).toMatchSnapshot();
  });

  test('should be able to change the drag modules', () => {
    expect(
      ModulePositionReducer(
        moduleInitialState,
        startDrag({
          id: 2,
          oldX: 2,
          oldY: 10,
        })
      )
    ).toMatchSnapshot();
  });
});
