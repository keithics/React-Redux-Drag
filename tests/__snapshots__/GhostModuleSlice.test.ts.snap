// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Module Tests should be able to change the drag modules 1`] = `
Array [
  Object {
    "coord": Object {
      "h": 200,
      "w": 2,
      "x": 1,
      "y": 80,
    },
    "id": 1,
  },
  Object {
    "coord": Object {
      "h": 100,
      "w": 2,
      "x": 5,
      "y": 180,
    },
    "id": 2,
  },
  Object {
    "coord": Object {
      "h": 200,
      "w": 3,
      "x": 4,
      "y": 290,
    },
    "id": 3,
  },
]
`;

exports[`Module Tests should be able to change the module coordinates 1`] = `
Array [
  Object {
    "coord": Object {
      "h": 200,
      "w": 2,
      "x": 1,
      "y": 80,
    },
    "id": 1,
  },
  Object {
    "coord": Object {
      "h": 100,
      "w": 2,
      "x": 1,
      "y": 2,
    },
    "id": 2,
  },
  Object {
    "coord": Object {
      "h": 200,
      "w": 3,
      "x": 4,
      "y": 290,
    },
    "id": 3,
  },
]
`;

exports[`Module Tests should return the initial state 1`] = `
Array [
  Object {
    "coord": Object {
      "h": 200,
      "w": 2,
      "x": 1,
      "y": 80,
    },
    "id": 1,
  },
  Object {
    "coord": Object {
      "h": 100,
      "w": 2,
      "x": 5,
      "y": 180,
    },
    "id": 2,
  },
  Object {
    "coord": Object {
      "h": 200,
      "w": 3,
      "x": 4,
      "y": 290,
    },
    "id": 3,
  },
]
`;
